src/main/kotlin/org/pkl/lsp/documentation/toMarkdown.kt

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.lsp.documentation
+
+import org.pkl.lsp.Project
+import org.pkl.lsp.ast.*
+import org.pkl.lsp.packages.dto.PklProject
+import org.pkl.lsp.resolvers.ResolveVisitors
+import org.pkl.lsp.resolvers.Resolvers
+import org.pkl.lsp.type.Type
+import org.pkl.lsp.type.computeExprType
+import org.pkl.lsp.type.computeResolvedImportType
+import org.pkl.lsp.type.computeThisType
+
+fun PklNode.toMarkdown(originalNode: PklNode?, context: PklProject?): String {
+  val markdown = doRenderMarkdown(originalNode, context)
+  return when {
+    this is PklModule && header != null -> showDocCommentAndModule(header, markdown, context)
+    else -> showDocCommentAndModule(this, markdown, context)
+  }
+}
+
+fun Type.toMarkdown(project: Project, context: PklProject?): String {
+  val markdown = render()
+  val ctx = getNode(project, context)
+  showDocCommentAndModule(ctx, markdown, context)
+  return when {
+    ctx is PklModule && ctx.header != null ->
+      showDocCommentAndModule(ctx.header!!, markdown, context)
+    else -> showDocCommentAndModule(ctx, markdown, context)
+  }
```

Comment:
That `when` seems more complicated than it needs to be. More important, though, is the `showDocCommentAndModule(ctx` duplication (when `ctx` is _not_ `PklModule`) intentional?
```suggestion
    return showDocCommentAndModule((ctx as? PklModule)?.header ?: ctx, markdown, context)
```